inductiva/structures/post_processing/fenicsx.py

```diff
@@ -0,0 +1,149 @@
+"""Post-process FEniCSx simulation outputs."""
+
+import os
+import tempfile
+
+import pyvista as pv
+
+import inductiva
+
+
+class FEniCSxSimulationOutput:
+    """Post-process FEniCSx simulation outputs."""
+
+    def __init__(self, sim_output_path: inductiva.types.Path):
+        """Initializes a 'FEniCSxSimulationOutput' object.
+
+        Args:
+            sim_output_path: Path to simulation output files.
+            """
+        self.sim_output_path = sim_output_path
+
+    def remove_time_values_from_xdmf(self, output_file_path: str) -> None:
+        """Removes Time Value lines from an XDMF file.
+
+        This method is crucial due to PyVista's limitations in inspecting time
+        values directly.
+        It reads an XDMF file, removes lines containing Time Values, and writes
+        the modified content to a new file.
+
+        Args: 
+            output_file_path: The path to save the modified XDMF content.
+        """
+
+        # Read the XDMF file
+        with open(self.sim_output_path, "r", encoding="utf-8") as file:
+            xdmf_content = file.readlines()
+
+        # Create an empty list to store the modified content
+        new_xdmf_content = []
+
+        # Iterate through the lines in the file
+        for line in xdmf_content:
+
+            # Check if the line contains a Time Value
+            if "<Time Value=" not in line:
+                new_xdmf_content.append(line)
+
+        # Write the modified content back to the output file
+        with open(output_file_path, "w", encoding="utf-8") as file:
+            file.writelines(new_xdmf_content)
+
+    @inductiva.utils.optional_deps.needs_structures_extra_deps
+    def render(self,
+               field_names: list = None,
+               scalar_bar: bool = True,
+               transparent_background: bool = True) -> None:
+        """Render the simulation fields as images.
+
+        Args:
+            field_names (list, optional): List of field names to render. Default
+              is None.
+            scalar_bar (bool, optional): Whether to include a scalar bar legend
+              in the visualization. Default is True.
+            transparent_background (bool, optional): Whether the background of
+              the saved images should be transparent. Default is True.
+        """
+
+        # Create a temporary directory
+        temp_dir = tempfile.mkdtemp()
+
+        # Specify the filename within the temporary directory
+        temp_file_path = os.path.join(temp_dir, "temp_results.xdmf")
+
+        # Remove time values from the XDMF file
+        self.remove_time_values_from_xdmf(self.sim_output_path)
+
+        # Read simulation output data
+        reader = pv.get_reader(temp_file_path)
```

Comment:
Imagine you have another rendering function or you call this function again, will you do this all the time? The information we are removing is it super important for the user to have, or can we just deliver the sim_output_path already without it?

Reply:
Yes, it's not crucial. With this function, we're simply removing lines that contain the phrase 'time values,' so we're not losing any data. The only downside is having to repeat this process

Reply:
Then just remove it once (even in the backend if you want) and do not repeat it - it can cause some unneeded crashes.